documentation/sphinx/source/restore-validation-testing.rst

```diff
@@ -0,0 +1,486 @@
+
+Phase 5: Verify Results
+------------------------
+
+Check Audit Status
+^^^^^^^^^^^^^^^^^^
+
+::
+
+    fdb> get_audit_status validate_restore id <AuditID>
+
+**Expected Output (Success)**::
+
+    Audit result is:
+    AuditStorageState: [ID]: <AuditID>, [Range]: ["","\\xff"), [Type]: 5, [Phase]: 2
+
+Where:
+
+- Type: 5 = ValidateRestore
+- Phase: 2 = Complete (no errors)
+
+**If Phase: 3 = Error**, there were validation failures!
+
+Check Trace Logs
+^^^^^^^^^^^^^^^^
+
+Look for validation events in the server logs::
+
+    grep "AuditRestore" ~/fdb_test_data/*.log | tail -20
+
+Look for:
+
+- ``SSAuditRestoreBegin`` - Validation started
+- ``SSAuditRestoreComplete`` - Validation finished successfully
+- ``SSAuditRestoreError`` - Validation found errors (check details!)
+
+Phase 6: Testing a Failed Audit
+--------------------------------
+
+To verify that the audit correctly detects mismatches, you can intentionally modify the source data and rerun the audit.
+
+Modify Source Data
+^^^^^^^^^^^^^^^^^^
+
+::
+
+    ~/build_output/bin/fdbcli -C ~/fdb_test.cluster
+
+    fdb> writemode on
+
+    # Modify one of the original source values to create a mismatch
+    fdb> set testkey1 "modified_value"
+
+    # Verify the change
+    fdb> get testkey1
+
+This creates a mismatch because:
+
+- Current source data: ``testkey1`` = ``modified_value`` (modified after backup)
+- Restored data: ``\xff\x02/rlog/testkey1`` = ``testvalue1`` (from backup)
+
+Run Audit Again
+^^^^^^^^^^^^^^^
+
+::
+
+    # Start a new validation audit
+    fdb> audit_storage validate_restore "" "\xff"
+
+Save the new Audit ID returned.
+
+Check for Expected Failure
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+::
+
+    # Monitor the audit status
+    fdb> get_audit_status validate_restore id <NewAuditID>
+
+**Expected Output (Failure)**::
+
+    Audit result is:
+    AuditStorageState: [ID]: <AuditID>, [Range]: ["","\\xff"), [Type]: 5, [Phase]: 3
+
+Where:
+
+- Phase: 3 = Error (validation found mismatches)
+
+Check Error Details
+^^^^^^^^^^^^^^^^^^^
+
+::
+
+    # Check detailed error information
+    fdb> get_audit_status validate_restore phase error
+
+    # Check trace logs for specific error details
+    grep "SSAuditRestoreError" ~/fdb_test_data/*.log | tail -20
+
+The logs should show which key had a mismatch and what the differing values were.
+
+Restore Correct Source Data for Next Tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+::
+
+    # Restore the original source value to match the backup
+    fdb> writemode on
+    fdb> set testkey1 testvalue1
+
+    # Verify the restoration
+    fdb> get testkey1
+
+Phase 7: Understanding Audit Design and Limitations
+----------------------------------------------------
+
+.. important::
+   The restore validation audit is designed to run immediately after a restore operation to verify the restore process didn't corrupt data. It compares the current database state with the restored data at the time of the audit.
```

Comment:
Why can't we lock the database before getting the restorableVersion? It's very difficult to track whether any updates happened to the cluster after we started restoring. And unnecessarily we will be debugging the audit errors